### PR TITLE
:broom: Just remove the tools, previously moved out to toolbox

### PR DESCRIPTION
**Pull request by cardil**
**Which issue(s) this PR fixes**:<br>

 - Relates to https://github.com/knative/community/issues/1291
 - Part of https://github.com/knative/infra/pull/51
 - Depends on https://github.com/knative/test-infra/pull/3784

### CODE REVIEW

**Comment by kvmware** [comment]
It seems the PR is removing `configgen` and `release-jobs-syncer`, are those going to live in toolbox? They are very infra specific.

**Comment by cardil** [comment]
> It seems the PR is removing `configgen` and `release-jobs-syncer`, are those going to live in toolbox? They are very infra specific.

Config gen is left. The `release-jobs-syncer` is in toolbox repo already: https://github.com/knative/toolbox/tree/main/release-jobs-syncer

File: tools/configgen/pkg/testgrid.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ import (
 	"sort"
 	"strings"
 
-	"knative.dev/test-infra/pkg/testgrid"
+	"knative.dev/test-infra/tools/configgen/pkg/testgrid"
```
Comment (kvmware) [inline tools/configgen/pkg/testgrid.go:28]: ?

**Comment by kvmware** [comment]
Hmm if `release-jobs-syncer` is in toolbox maybe configgen should be too? or `release-jobs-syncer` should be moved back into infra for consistency sake one or the other?

File: tools/configgen/pkg/testgrid.go (hunk 1 above)
Comment (cardil) [reply tools/configgen/pkg/testgrid.go:28]: The configgen is using the `./pkg/testgrid` package. I'm also removing the whole `./pkg` package, so moving those two files to the `configgen` directory makes sense to me.

**Comment by cardil** [comment]
> Hmm if `release-jobs-syncer` is in toolbox maybe configgen should be too? or `release-jobs-syncer` should be moved back into infra for consistency sake one or the other?

Hmm, I moved the r-j-s to toolbox as it uses a lot more of the common toolbox/pkg, and has it's own go.mod file with plenty of deps.

For sure we need to adjust this: https://github.com/knative/toolbox/blob/292433b19b3627acfb6f52077c47e9dd796b378a/release-jobs-syncer/main.go#L34-L35 (or make it configurable)

**Comment by kvmware** [comment]
```
 ---> Running in 47fa5e975a3a
stat /go/src/knative.dev/test-infra/tools/kntest/cmd/kntest: directory not found
              
The command '/bin/sh -c cd /go/src/knative.dev/test-infra && go install ./tools/kntest/cmd/kntest' returned a non-zero code: 1
ERROR
ERROR: build step 0 "gcr.io/cloud-builders/docker" failed: step exited with non-zero status: 1
```
but https://github.com/knative/test-infra/tree/main/tools/kntest/cmd/kntest very much does exist... ?

**Comment by cardil** [comment]
> but https://github.com/knative/test-infra/tree/main/tools/kntest/cmd/kntest very much does exist... ?

It is just removed. But the Dockerfiles aren't changed. I changed them in #53 - and it trieds to build it as `test-infra`, not `infra`.

I don't know if we can land it without a complete set of changes, ie.: #53 (or at least some of them)

**Comment by kvmware** [comment]
> > Hmm if `release-jobs-syncer` is in toolbox maybe configgen should be too? or `release-jobs-syncer` should be moved back into infra for consistency sake one or the other?
> 
> Hmm, I moved the r-j-s to toolbox as it uses a lot more of the common toolbox/pkg, and has it's own go.mod file with plenty of deps.
> 
> For sure we need to adjust this: https://github.com/knative/toolbox/blob/292433b19b3627acfb6f52077c47e9dd796b378a/release-jobs-syncer/main.go#L34-L35 (or make it configurable)

Did you capture this in an issue somewhere?

**Comment by kvmware** [comment]
> > but https://github.com/knative/test-infra/tree/main/tools/kntest/cmd/kntest very much does exist... ?
> 
> It is just removed, because it is building from THIS repo (named as `test-infra`). But the Dockerfiles aren't changed. I changed them in #53 - and it tries to build it as `test-infra`, not `infra`.
> 
> I don't know if we can land it without a complete set of changes, ie.: #53 (or at least some of them)

when I retested it seemed to have passed. So, it seems that it was a fluke

**Comment by cardil** [comment]
> > For sure we need to adjust this: https://github.com/knative/toolbox/blob/292433b19b3627acfb6f52077c47e9dd796b378a/release-jobs-syncer/main.go#L34-L35 (or make it configurable)
> 
> Did you capture this in an issue somewhere?

Good point. Here: https://github.com/knative/toolbox/issues/6

**Comment by cardil** [comment]
It isn't a fluke. It failed again: https://prow.knative.dev/view/gs/knative-prow/pr-logs/pull/knative_infra/52/pull-knative-infra-prow-tests-image-build/1643616125075853312

**Comment by kvmware** [comment]
Could you bring in the change that gets this to pass. Please make it it's own commit message so it is easy to review. Also, fix the `THIS repo (named as test-infra)` to be named correctly.

**Comment by cardil** [comment]
> to be named correctly

That's basically what #53 does.

**Comment by kvmware** [comment]
/lgtm

Merging this in to get to next PR since there is not much other than removes in this PR

**Pull request merged**